File: connext-backend/internal/models/invitation.go
// internal/models/invitation.go
package models

import "gorm.io/gorm"

// Invitation model for network sharing
type Invitation struct {
	gorm.Model

	// Who sent the invitation
	InviterID uint `gorm:"not null" json:"inviter_id"`

	// Who is invited (email)
	InviteeEmail string `gorm:"size:100;not null" json:"invitee_email"`
	InviteeName  string `gorm:"size:50" json:"invitee_name"`

	// Invitation details
	Token  string `gorm:"size:64;uniqueIndex;not null" json:"token"`
	Status string `gorm:"size:20;default:'pending'" json:"status"` // pending, accepted, declined, cancelled

	// If accepted, who accepted it
	AcceptedByID *uint `json:"accepted_by_id,omitempty"`
}

File: connext_desktop/lib/models/invitation.dart
// lib/models/invitation.dart
class Invitation {
  final int id;
  final String inviteeEmail;
  final String? inviteeName;
  final String status;
  final String token;
  final DateTime createdAt;
  
  Invitation({
    required this.id,
    required this.inviteeEmail,
    this.inviteeName,
    required this.status,
    required this.token,
    required this.createdAt,
  });
  
  factory Invitation.fromJson(Map<String, dynamic> json) {
    return Invitation(
      id: json['id'] as int,
      inviteeEmail: json['invitee_email'] as String,
      inviteeName: json['invitee_name'] as String?,
      status: json['status'] as String,
      token: json['token'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}